packages/vue/src/utils/useClerkLoaded.ts

```diff
@@ -0,0 +1,31 @@
+import type { LoadedClerk } from '@clerk/types';
+import { watch } from 'vue';
+
+import { useClerk } from '../composables';
+
+/**
+ * Executes a callback when Clerk is loaded.
+ *
+ * @param callback - Function to execute once Clerk is loaded
+ * @example
+ * ```ts
+ * useClerkLoaded((clerk) => {
+ *   clerk.redirectToSignUp(props);
+ * });
+ * ```
+ */
+export const useClerkLoaded = (callback: (clerk: LoadedClerk) => void) => {
+  const clerk = useClerk();
+
+  watch(
+    clerk,
+    unwrappedClerk => {
+      if (!unwrappedClerk?.loaded) {
+        return;
+      }
+
+      callback(unwrappedClerk as LoadedClerk);
+    },
+    { immediate: true },
+  );
+};
```

Comment:
This composable is used inside the control components where we want to execute a Clerk method when Clerk is loaded.

We're using a `watch` here instead of `watchEffect` to react to a specific state, instead of watching every reactivate state inside the callback